<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>937</width>
    <height>623</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SLAM Control Center</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="grpTuners">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>20</y>
      <width>161</width>
      <height>511</height>
     </rect>
    </property>
    <property name="title">
     <string>Tuners</string>
    </property>
    <widget class="QSlider" name="sldrRoll">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>29</width>
       <height>160</height>
      </rect>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
     <property name="sliderPosition">
      <number>5000</number>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="spnRoll">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>240</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="decimals">
      <number>4</number>
     </property>
     <property name="singleStep">
      <double>0.005000000000000</double>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtRollMin">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>220</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>-.03</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtRollMax">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>40</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>.03</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtPitchMin">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>220</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>-.03</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="spnPitch">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>240</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="decimals">
      <number>4</number>
     </property>
     <property name="singleStep">
      <double>0.005000000000000</double>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtPitchMax">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>40</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>.03</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtYawMin">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>460</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>-0.5</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="spnYaw">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>480</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="decimals">
      <number>4</number>
     </property>
     <property name="singleStep">
      <double>0.005000000000000</double>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtYawMax">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>280</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>0.5</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtThrustMin">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>460</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>.48</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="spnThrust">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>480</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="decimals">
      <number>4</number>
     </property>
     <property name="singleStep">
      <double>0.005000000000000</double>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtThrustMax">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>280</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>0.52</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>260</y>
       <width>51</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Yaw</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>260</y>
       <width>51</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Thrust</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>51</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Roll</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>20</y>
       <width>51</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Pitch</string>
     </property>
    </widget>
    <widget class="QSlider" name="sldrPitch">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>60</y>
       <width>29</width>
       <height>160</height>
      </rect>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
     <property name="sliderPosition">
      <number>5000</number>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
    <widget class="QSlider" name="sldrYaw">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>300</y>
       <width>29</width>
       <height>160</height>
      </rect>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
     <property name="sliderPosition">
      <number>5000</number>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
    <widget class="QSlider" name="sldrThrust">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>300</y>
       <width>29</width>
       <height>160</height>
      </rect>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
     <property name="sliderPosition">
      <number>5000</number>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="grpControl">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>440</y>
      <width>441</width>
      <height>80</height>
     </rect>
    </property>
    <property name="title">
     <string>Controls</string>
    </property>
    <widget class="QPushButton" name="btnStop">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>50</y>
       <width>99</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Stop SLAM</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnHome">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>20</y>
       <width>99</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Go Home</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnStart">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>171</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>START</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnTakeoff">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>50</y>
       <width>99</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Takeoff</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnArm">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>20</y>
       <width>99</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>ARM</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnInit">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>171</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Init SLAM</string>
     </property>
    </widget>
   </widget>
   <widget class="MyGLWidget" name="myGLWidget" native="true">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>421</width>
      <height>391</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="lblStatus">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>560</y>
      <width>391</width>
      <height>17</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::WinPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="text">
     <string>Status</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblMode">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>560</y>
      <width>421</width>
      <height>20</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::WinPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="text">
     <string>Mode</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>40</y>
      <width>291</width>
      <height>211</height>
     </rect>
    </property>
    <property name="title">
     <string>Gain</string>
    </property>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>21</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>P</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>21</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>I</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>21</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>D</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="spnP">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>20</y>
       <width>241</width>
       <height>51</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>15</pointsize>
      </font>
     </property>
     <property name="decimals">
      <number>8</number>
     </property>
     <property name="maximum">
      <double>2.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.002000000000000</double>
     </property>
     <property name="value">
      <double>0.080000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="spnI">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>80</y>
       <width>241</width>
       <height>51</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>15</pointsize>
      </font>
     </property>
     <property name="decimals">
      <number>8</number>
     </property>
     <property name="maximum">
      <double>0.500000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.002000000000000</double>
     </property>
     <property name="value">
      <double>0.020000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="spnD">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>140</y>
       <width>241</width>
       <height>51</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>15</pointsize>
      </font>
     </property>
     <property name="decimals">
      <number>8</number>
     </property>
     <property name="maximum">
      <double>0.500000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.002000000000000</double>
     </property>
     <property name="value">
      <double>0.020000000000000</double>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="btnHover">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>520</y>
      <width>171</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Hover</string>
    </property>
   </widget>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGLWidget</class>
   <extends>QWidget</extends>
   <header>myglwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
